apis/python/src/tiledbsoma/_experiment.py

```diff
@@ -74,3 +98,26 @@ def _set_create_metadata(cls, handle: Wrapper[Any]) -> None:
         # TileDB Cloud UI to detect that they are SOMA datasets.
         handle.metadata["dataset_type"] = "soma"
         return super()._set_create_metadata(handle)
+
+    # in tiledbsoma
+    def axis_query(  # type: ignore
+        self,
+        measurement_name: str,
+        *,
+        obs_query: Optional[query.AxisQuery] = None,
+        var_query: Optional[query.AxisQuery] = None,
+    ):
+        """Creates an axis query over this experiment.
+
+
+        Lifecycle: maturing
+        """
+        # mypy doesn't quite understand descriptors so it issues a spurious
+        # error here.
+        return query.ExperimentAxisQuery(  # type: ignore
+            self,
+            measurement_name,
+            obs_query=obs_query or query.AxisQuery(),
+            var_query=var_query or query.AxisQuery(),
+            index_factory=_indexer_map_locations,  # comment this line to disable thw C++ indexer
```

Comment:
With [the context in the somacore review](https://github.com/single-cell-data/SOMA/pull/181#discussion_r1422604804) for why we want to have this still just take a `NDArray` as its only parameter, we can still pass the context through to the index builder:

```python
def indexer_map_locations(keys: ..., ctx: ...) -> clib.IntIndexer:
  ...

class Experiment(...):
  ...

  def axis_query(self, ...):
    return query.ExperimentAxisQuery(
        ...,
        index_factory=functools.partial(indexer_map_locations, context=self.context),
    )
```

This means that the `indexer_map_locations` function still knows about the `self.context`, but the somacore code doesn’t need to plumb it around just to pass back into our code.